components/browser/engine-gecko/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt

```diff
@@ -271,6 +272,7 @@ class GeckoEngineSessionTest {
     // This will be addressed on https://github.com/mozilla-mobile/android-components/issues/8312
     @Suppress("deprecation")
     @Test
+    @Ignore("InvalidUseOfMatchersException")
```

Comment:
@csadilek I have no clue why Mockito is complaining here. The use of matchers looks correct?

Reply:
@pocmo ah yeah, the root cause is:

```
Also, this error might show up because you use argument matchers with methods that cannot be mocked.
Following methods *cannot* be stubbed/verified: final/private/equals()/hashCode().
Mocking methods declared on non-public parent classes is not supported.
```

Reply:
@pocmo ok this took me a while, but I found it. It turns out ordering matters when using named arguments as mockito will otherwise return null for that argument.


So this can be fixed by correcting the order of the cookie and userAgent parameter. This will pass:

```
verify(observer).onExternalResource(
            url = eq("https://download.mozilla.org/image.png"),
            fileName = eq("image.png"),
            contentLength = eq(42),
            contentType = eq("image/png"),
            cookie = eq(null),
            userAgent = eq(null),
            isPrivate = eq(true),
            response = captor.capture()
        )
```

Reply:
Ah, great find! Thank you :)